app/src/main/java/ciserver/PipelinePuller.java

```diff
@@ -0,0 +1,48 @@
+package ciserver;
+
+import java.io.File;
+import java.io.IOException;
+
+import org.apache.commons.io.FileUtils;
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.errors.GitAPIException;
+
+public class PipelinePuller implements PipelineComponent {
+	/**
+	 * Pulls the repository and checks out the head_commit
+	 * 
+	 * Inspiration from:
+	 * https://github.com/centic9/jgit-cookbook/blob/master/src/main/java/org/dstadler/jgit/unfinished/PullRemoteRepository.java
+	 * Can only pull from public repositories.
+	 * 
+	 * @return the status of the pull action
+	 */
+	public PipelineStatus execute(String pipelineDir, PushEvent event) {
+		// Use the head_commit id as name for the repository directory
+		String directoryPath = String.format("%s/repositories/%s", pipelineDir, event.headCommit.id);
```

Comment:
```suggestion
		String directoryPath = String.format("%s/%s/%s", pipelineDir, event.repository.name, event.headCommit.id);
```
It's probably good to also have the name of the repository in the path if you have multiple repositories.

Reply:
You could, but the probability of hash collisions are so small, it shouldn't be a problem.